lib/shopify_app/middleware/same_site_cookie_middleware.rb

```diff
@@ -10,9 +10,13 @@ def call(env)
       status, headers, body = @app.call(env)
       user_agent = env['HTTP_USER_AGENT']
 
+      @request = Rack::Request.new(env) 
+      is_https = @request.env['HTTPS'] == 'on' || @request.env['HTTP_X_SSL_REQUEST'] == 'on'
```

Comment:
You can check this by simply doing `Rack::Request.new(env).ssl?`

Reply:
+1